examples/fp_delegation.cpp

```diff
@@ -123,22 +123,27 @@ class delegating_sender {
     }
   };
 
+  using SchedSender =
+      schedule_result_t<get_scheduler_result_t<timed_single_thread_context&>>;
+
+  template <typename Receiver>
+  using operation =
+      delegating_operation<
+          connect_result_t<
+              schedule_result_t<get_scheduler_result_t<Receiver>>,
+              Receiver>,
+          LocalContextType<connect_result_t<SchedSender, Receiver>>>;
+
   template <typename Receiver>
-  auto connect(Receiver&& receiver) {
+  operation<Receiver> connect(Receiver&& receiver) {
```

Comment:
It seems that attempting to use `auto` as the return-type of `connect()` always runs into trouble with recursive instantiation of some sort.

Reply:
Yes. Every time we've done that it's resulted in hours of hair pulling tracking down instantiation problems. Let's never do that. And we really need to get in the habit of constraining `connect` at the very least.